velox/dwio/common/tests/LoggedExceptionTest.cpp

```diff
@@ -13,13 +13,60 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <folly/Random.h>
 #include <gtest/gtest.h>
 #include "velox/dwio/common/exception/Exception.h"
 
 using namespace facebook::velox::dwio::common::exception;
 using namespace facebook::velox;
 
 namespace {
+void testTraceCollectionSwitchControl(bool enabled) {
+  // Logged exception is system type of velox exception.
+  // Disable rate control in the test.
+  FLAGS_velox_exception_stacktrace_rate_limit_ms = 0;
+  FLAGS_velox_exception_usr_stacktrace_rate_limit_ms = 0;
+  FLAGS_velox_exception_sys_stacktrace_rate_limit_ms = 0;
+
+  /// Test the old flag to deprecate.
+  FLAGS_velox_exception_stacktrace = enabled ? true : false;
+  /// NOTE: the user flag should not affect the tracing behavior of system type
+  /// of exception collection.
+  FLAGS_velox_exception_usr_stacktrace_enabled = folly::Random::oneIn(2);
```

Comment:
Using random like this makes it hard to reproduce failures. Is this needed?

Reply:
Yeah, but we logged out the actual flag value used when the test fails?